packages/synapse-react-client/src/lib/containers/auth/Login.tsx

```diff
@@ -0,0 +1,211 @@
+import * as React from 'react'
+import { Form } from 'react-bootstrap'
+import ButtonWithIcon from '../../assets/ButtonWithIcon'
+import { SynapseClient } from '../../utils'
+import {
+  getEndpoint,
+  BackendDestinationEnum,
+} from '../../utils/functions/getEndpoint'
+import { GoogleIcon24 } from '../../assets/GoogleIcon24'
+import { Button, IconButton, Link } from '@mui/material'
+import IconSvg from '../IconSvg'
+import FullWidthAlert from '../FullWidthAlert'
+import { useState } from 'react'
+
+export const PROVIDERS = {
+  GOOGLE: 'GOOGLE_OAUTH_2_0',
+  ORCID: 'ORCID',
+}
+
+type Props = {
+  ssoRedirectUrl?: string
+  sessionCallback: () => void // Callback is invoked after login
+  registerAccountUrl?: string
+  resetPasswordUrl?: string
+  handleIsOnUsernameOrPasswordScreen?: React.Dispatch<
+    React.SetStateAction<boolean | undefined>
+  >
+  showUsernameOrPassword?: boolean | undefined
+  renderBackButton?: boolean
+}
+
+/**
+ *  Demo of user session, show login screen and handling user login submission.
+ *
+ *  To support Google SSO in your portal, you must add your domain to the Authorized Redirect URIs
+ *  for Synapse authentication.
+ *  This can be done by visiting https://sagebionetworks.jira.com/servicedesk/customer/portal/9 to set up a collaboration.
+ *  Synapse engineers must add your redirect URL in the Google API console found at https://console.cloud.google.com/ for this functionality to work.
+ */
+function Login(props: Props) {
```

Comment:
My motivation for converting this to a function component is that I want to use a new custom hook to manage the login state as it grows increasingly complex. Using all function components will make it easier to integrate this custom hook into existing components.